File: backend/app/schemas/documents.py
from uuid import UUID

from pydantic import BaseModel


class DocumentTemplateResponse(BaseModel):
    id: UUID
    name: str
    category: str
    description: str
    jurisdiction: str
    variables_schema: dict

    class Config:
        from_attributes = True


class DocumentGenerateRequest(BaseModel):
    template_id: UUID
    variables: dict


class DocumentGenerateResponse(BaseModel):
    id: UUID
    title: str
    content: str
    file_url: str | None

    class Config:
        from_attributes = True

File: frontend/src/types/documents.ts
export interface DocumentTemplate {
  id: string;
  name: string;
  category: string;
  description: string;
  jurisdiction: string;
  variables_schema: Record<string, unknown>;
}

export interface GenerateRequest {
  template_id: string;
  variables: Record<string, string>;
}

export interface GeneratedDocument {
  id: string;
  title: string;
  content: string;
  file_url: string | null;
}
